packages/core/src/useFormField/useFormField.ts

```diff
@@ -36,14 +37,21 @@ export type WithFieldProps<TControlProps extends object> = Simplify<
   }
 >;
 
-export function useFormField<TValue = unknown>(init?: FormFieldInit<TValue>): FormField<TValue> {
+export function useFormField<TValue = unknown>(
+  init?: FormFieldInit<TValue>,
+  constrolType: string = 'Field',
+): FormField<TValue> {
   const controllerProps = normalizeProps(init ?? { label: '' });
   const state = useFieldState<TValue>(init);
   const controller = useFieldController({
     ...controllerProps,
     errorMessage: state.errorMessage,
   });
 
+  if (__DEV__) {
+    registerField(state, controller.controlType ?? constrolType);
+  }
+
```

Comment:
The `controlType` can be provided by either the field `controller` (most fields), or passed directly (like `useHiddenField`)